plugins/module_utils/idrac_redfish.py

```diff
@@ -41,14 +41,16 @@
 
 idrac_auth_params = {
     "idrac_ip": {"required": True, "type": 'str'},
-    "idrac_user": {"required": True, "type": 'str', "fallback": (env_fallback, ['IDRAC_USERNAME'])},
-    "idrac_password": {"required": True, "type": 'str', "aliases": ['idrac_pwd'], "no_log": True, "fallback": (env_fallback, ['IDRAC_PASSWORD'])},
+    "idrac_user": {"required": False, "type": 'str', "fallback": (env_fallback, ['IDRAC_USERNAME'])},
+    "idrac_password": {"required": False, "type": 'str', "aliases": ['idrac_pwd'], "no_log": True, "fallback": (env_fallback, ['IDRAC_PASSWORD'])},
+    "x_auth_token": {"required": False, "type": 'str', "no_log": True, "fallback": (env_fallback, ['IDRAC_X_AUTH_TOKEN'])},
     "idrac_port": {"required": False, "default": 443, "type": 'int'},
     "validate_certs": {"type": "bool", "default": True},
     "ca_path": {"type": "path"},
     "timeout": {"type": "int", "default": 30},
-
 }
+auth_mutually_exclusive = [["idrac_user", "x_auth_token"], ["idrac_password", "x_auth_token"]]
+auth_required_together = [["idrac_user", "idrac_password"]]
```

Comment:
```suggestion
auth_required_together = [("idrac_user", "idrac_password")]
```
Can you check the syntax i think it should a list of tuple.  similarly for other files.